ige/src/plugin/input/utils.cpp

```diff
@@ -0,0 +1,288 @@
+    { KeyboardKey::KEY_F7, "KEY_F7" },
+    { KeyboardKey::KEY_F8, "KEY_F8" },
+    { KeyboardKey::KEY_F9, "KEY_F9" },
+    { KeyboardKey::KEY_G, "KEY_G" },
+    { KeyboardKey::KEY_H, "KEY_H" },
+    { KeyboardKey::KEY_I, "KEY_I" },
+    { KeyboardKey::KEY_INSERT, "KEY_INSERT" },
+    { KeyboardKey::KEY_J, "KEY_J" },
+    { KeyboardKey::KEY_K, "KEY_K" },
+    { KeyboardKey::KEY_L, "KEY_L" },
+    { KeyboardKey::KEY_M, "KEY_M" },
+    { KeyboardKey::KEY_N, "KEY_N" },
+    { KeyboardKey::KEY_O, "KEY_O" },
+    { KeyboardKey::KEY_P, "KEY_P" },
+    { KeyboardKey::KEY_PAGE_DOWN, "KEY_PAGE_DOWN" },
+    { KeyboardKey::KEY_PAGE_UP, "KEY_PAGE_UP" },
+    { KeyboardKey::KEY_Q, "KEY_Q" },
+    { KeyboardKey::KEY_R, "KEY_R" },
+    { KeyboardKey::KEY_S, "KEY_S" },
+    { KeyboardKey::KEY_SHIFT_LEFT, "KEY_SHIFT_LEFT" },
+    { KeyboardKey::KEY_SHIFT_RIGHT, "KEY_SHIFT_RIGHT" },
+    { KeyboardKey::KEY_SPACE, "KEY_SPACE" },
+    { KeyboardKey::KEY_T, "KEY_T" },
+    { KeyboardKey::KEY_TAB, "KEY_TAB" },
+    { KeyboardKey::KEY_U, "KEY_U" },
+    { KeyboardKey::KEY_V, "KEY_V" },
+    { KeyboardKey::KEY_W, "KEY_W" },
+    { KeyboardKey::KEY_X, "KEY_X" },
+    { KeyboardKey::KEY_Y, "KEY_Y" },
+    { KeyboardKey::KEY_Z, "KEY_Z" },
+};
+
+static std::unordered_map<MouseButton, std::string> MOUSE_BUTTON_TO_STRING = {
+    { MouseButton::LEFT, "LEFT" },
+    { MouseButton::RIGHT, "RIGHT" },
+    { MouseButton::MIDDLE, "MIDDLE" },
+};
+
+static std::unordered_map<std::string, MouseButton> STRING_TO_MOUSE_BUTTON = {
+    { "LEFT", MouseButton::LEFT },
+    { "RIGHT", MouseButton::RIGHT },
+    { "MIDDLE", MouseButton::MIDDLE },
+};
+
+std::optional<KeyboardKey>
+InputsFromString::keyboard_key_from_string(const std::string& str)
+{
+    auto key = STRING_TO_KEYBOARD_KEY.find(str);
+
+    if (key == STRING_TO_KEYBOARD_KEY.end()) {
+        return {};
+    }
+    return key->second;
+}
+
+std::optional<ControllerButton>
+InputsFromString::controller_button_from_string(const std::string& str)
+{
+    auto button = STRING_TO_CONTROLLER_BUTTON.find(str);
+
+    if (button == STRING_TO_CONTROLLER_BUTTON.end()) {
+        return {};
+    }
+    return button->second;
+}
+
+std::optional<MouseButton>
+InputsFromString::mouse_button_from_string(const std::string& str)
+{
+    auto button = STRING_TO_MOUSE_BUTTON.find(str);
+
+    if (button == STRING_TO_MOUSE_BUTTON.end()) {
+        return {};
+    }
+    return button->second;
+}
+
+std::optional<ControllerAxis>
+InputsFromString::controller_axis_from_string(const std::string& str)
+{
+    auto axis = STRING_TO_CONTROLLER_AXIS.find(str);
+
+    if (axis == STRING_TO_CONTROLLER_AXIS.end()) {
+        return {};
+    }
+    return axis->second;
+}
+
+std::string InputsToString::keyboard_key_to_string(const KeyboardKey& key)
+{
+    return KEYBOARD_KEY_TO_STRING[key];
+}
+
+std::string
+InputsToString::controller_button_to_string(const ControllerButton& button)
+{
+    return CONTROLLER_BUTTON_TO_STRING[button];
```

Comment:
use `std::unordered_map::at` instead of `operator[]`, because `operator[]` mutates the map (which is currently only possible because it's a global *variable*, which should be changed to a constant as mentioned in my previous comments)